### PR TITLE
feat(ui): artifact and configuration edit

### PR DESCRIPTION
**Pull request by ruanyl**
### Description
This PR adds artifact and configuration to version details page

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/model/model_details_panel.tsx [hunk 1]
```diff
@@ -24,14 +24,10 @@ import { generatePath, useHistory } from 'react-router-dom';
...
-import {
-  ErrorCallOut,
-  MODEL_NAME_FIELD_DUPLICATE_NAME_ERROR,
-  ModelDescriptionField,
-  ModelNameField,
-} from '../common';
+import { ErrorCallOut, ModelDescriptionField, ModelNameField } from '../common';
 
 import { BottomFormActionBar } from './bottom_form_action_bar';
+import { MODEL_NAME_FIELD_DUPLICATE_NAME_ERROR } from '../common/forms/form_constants';
```
Comment (wanglam) [inline public/components/model/model_details_panel.tsx:30]: Could we move this import before `./`? That's my person import order
```tsx
// import from "node_modules"
// import from "parent relative paths"
// import from "current directory"

```

File: public/components/model_version/model_version.tsx [hunk 2]
```diff
@@ -73,9 +73,12 @@ export const ModelVersion = () => {
           { key: 'Precision', value: '0.64', type: 'number' as const },
           { key: 'Task', value: 'Image classification', type: 'string' as const },
         ], // TODO: read from model.tags
-        configuration: JSON.stringify(model.model_config),
+        configuration: JSON.stringify(model.model_config, undefined, 2),
         modelFileFormat: model.model_format,
         // TODO: read model url or model filename
+        artifactSource: 'source_not_changed',
+        // modelFile: new File([], 'artifact.zip'),
```
Comment (wanglam) [inline public/components/model_version/model_version.tsx:80]: Do we need to construct the model file in the API integration phase?

File: public/components/model_version/__tests__/version_artifact.test.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+  it('should select "Keep existing" by default after click "Edit" button', async () => {
+    const user = userEvent.setup();
+    render(<TestApp />);
+
+    await user.click(screen.getByLabelText('edit version artifact'));
+    expect(screen.getByLabelText('Keep existing')).toBeChecked();
+    // model url and model file format are readonly
+    expect(screen.getByDisplayValue('http://url.to/artifact.zip')).toHaveAttribute('readonly');
+    expect(screen.getByDisplayValue('Torchscript(.pt)')).toHaveAttribute('readonly');
```
Comment (wanglam) [inline public/components/model_version/__tests__/version_artifact.test.tsx:73]: From my perspective, the model file format should be editable. Then user can modify the model file format without re-upload model. Let's check with designer about that. Won't block PR approve.

File: public/components/register_model/model_configuration.tsx [hunk 4]
```diff
@@ -3,101 +3,13 @@
...
-    const config = JSON.parse(value.trim());
-    if ('framework_type' in config && typeof config.framework_type !== 'string') {
-      return 'framework_type must be a string';
-    }
-  } catch {
-    return true;
-  }
-  return true;
-}
+import { ModelConfiguration } from '../common/forms/model_configuration';
```
Comment (wanglam) [inline public/components/register_model/model_configuration.tsx:10]: Do we need to expose these form components and constants to common folder?

File: public/components/model_version/__tests__/version_artifact.test.tsx (hunk 3 above)
Comment (ruanyl) [reply public/components/model_version/__tests__/version_artifact.test.tsx]: You're right, PR updated

File: public/components/register_model/model_configuration.tsx (hunk 4 above)
Comment (ruanyl) [reply public/components/register_model/model_configuration.tsx:10]: You mean components like `ModelConfiguration`?

File: public/components/model_version/model_version.tsx (hunk 2 above)
Comment (ruanyl) [reply public/components/model_version/model_version.tsx:80]: I'm thinking to create an empty File instance with the model file name we get from API. What do you think?

File: public/components/model/model_details_panel.tsx (hunk 1 above)
Comment (ruanyl) [reply public/components/model/model_details_panel.tsx:30]: Sounds good, we can even have editor import sort plugin for it. I will update it manually for now.

File: public/components/common/forms/__tests__/model_configuration/model_configuration.test.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
+/*
```
[the remark this answers is not in the document]
Comment (ruanyl) [reply public/components/common/forms/__tests__/model_configuration/model_configuration.test.tsx:1]: Right, let me update that.

File: public/components/model_version/model_version.tsx (hunk 2 above)
Comment (wanglam) [reply public/components/model_version/model_version.tsx:80]: I think that's make sense to init an empty file.

File: public/components/register_model/model_configuration.tsx (hunk 4 above)
Comment (wanglam) [reply public/components/register_model/model_configuration.tsx:10]: I mean expose `common/forms/model_configuration` contents to `common/forms/index.ts`. Then we can use `import {ModelConfiguration} from '../common'` to import this component. That's not a big issue.

**Review by wanglam — APPROVED** [review]

**Pull request merged**